File: WasteVisionWebFE/src/core/domain/Prediction.ts
import {  z } from "zod";
import { RoboflowModelSchema } from "./RoboflowModel";
import { UserSchema } from "./User";


export const PredictionSchema = z.object({
  id: z.string().optional(),
  userId: z.string(),
  modelId: z.string(),
  date: z.string(),
  originalImageUrl: z.string().url(),
  processedImageUrl: z.string().url(),
  roboflowModel: RoboflowModelSchema.optional(),
  user: UserSchema.optional(),
});

export type Prediction = z.infer<typeof PredictionSchema>;

File: WasteVisionWebBE/Domain/Prediction/Prediction.cs
using System;
using DDDSample1.Domain.Shared;
using DDDSample1.Domain.Users;
using DDDSample1.Domain.RoboflowModels;
using DDDSample1.Domain.ObjectPredictions;
using System.Collections.Generic;


namespace DDDSample1.Domain.Predictions
{
    public class Prediction : Entity<PredictionId>, IAggregateRoot
    {
        public UserId UserId { get; private set; }

        public User User { get; private set; }
        public RoboflowModelId RoboflowModelId { get; private set; }
        public RoboflowModel RoboflowModel { get; private set; }
        public Url OriginalImageUrl { get; private set; }
        public Url ProcessedImageUrl { get; private set; }
        public Date Date { get; private set; }

        public ICollection<ObjectPrediction> ObjectPredictions { get; private set; } = new List<ObjectPrediction>();

        private Prediction() { }

        public Prediction(UserId userId, RoboflowModelId roboflowModelIdId, Url originalImageUrl, Url processedImageUrl)
        {
            Id = new PredictionId(Guid.NewGuid());
            UserId = userId;
            RoboflowModelId = roboflowModelIdId;
            OriginalImageUrl = originalImageUrl;
            ProcessedImageUrl = processedImageUrl;
            Date = new Date(DateTime.UtcNow);
        }

    }
}